### ISSUE TITLE
Dynamic CLI support for windows OS

### ISSUE THREAD

**Issue by guptaaastha** [IndianOpenSourceFoundation/dynamic-cli#94]
Currently Dynamic CLI has a dependency on the `termios` module. This module does not support windows OS (refer the links provided below for reference). This issue is to point attention towards this drawback and open up a conversation whether this is something that can be addressed.

Please refer:
[https://docs.python.org/3/library/tty.html](https://docs.python.org/3/library/tty.html)
[https://docs.python.org/3/library/termios.html#module-termios](https://docs.python.org/3/library/termios.html#module-termios)

**Comment by NamamiShanker** [comment]
This dependency has been caused because of the `simple-term-menu` module. We need this module for StackExchange results to be printed the way they are. Please refer to PR #89. Windows users can use WSL for using Dynamic-CLI.

**Comment by guptaaastha** [comment]
One can surely use WSL but that would make Dynamic CLI restrictive and available only for users who are on Unix OS. I think we should try to be more inclusive and find alternatives that support all users regardless of their OS 😄

**Comment by NamamiShanker** [comment]
We can use `curses` but that can reduce UI/UX capabilities.

**Comment by pranavbaburaj** [comment]
> We can use `curses` but that can reduce UI/UX capabilities.

Hmm yeah

**Comment by pranavbaburaj** [comment]
Right now, I am planning to use an online linux environment like [repl.it](https://repl.it/) or [CS50 IDE](https://ide.cs50.io/).

**Comment by GouravSardana** [comment]
I think we can do specific change only for windows env. So that other would not face UX issues.

**Comment by GouravSardana** [comment]
Or else we need to find out some other way to support all the environment equally and efficiently.

**Comment by pranavbaburaj** [comment]
> Or else we need to find out some other way to support all the environment equally and efficiently.

I think this is what we have to do

**Comment by GouravSardana** [comment]
I'll do a research and find any alternative. You guy's can also search, If you find anything do let me know here.

**Comment by GouravSardana** [comment]
This is quite useful. You guys can also take a look. 2-3 people gave suggestions regarding the same problem that we are facing - https://github.com/bchao1/bullet/issues/2

**Comment by pranavbaburaj** [comment]
> This is quite useful. You guys can also take a look. 2-3 people gave suggestions regarding the same problem that we are facing - [bchao1/bullet#2](https://github.com/bchao1/bullet/issues/2)

But, they don't have a solution for that 🙁

**Comment by GouravSardana** [comment]
They have but that was specific to windows only. Still searching for some other alternative

**Comment by guptaaastha** [comment]
Let's be on the lookout for solutions

**Comment by pranavbaburaj** [comment]
https://github.com/IndianOpenSourceFoundation/dynamic-cli/discussions/97

**Comment by GouravSardana** [comment]
@pranavbaburaj i think you wrote wrong. It's not support for windows and by mistake you wrote it only supports for windows

**Comment by guptaaastha** [comment]
> @pranavbaburaj i think you wrote wrong. It's not support for windows and by mistake you wrote it only supports for windows

I just noticed this too, @pranavbaburaj  please correct it.

**Comment by pranavbaburaj** [comment]
I have updated it, Thank you 🙂

**Comment by GouravSardana** [comment]
We can use this as a backup option for windows- https://github.com/aegirhall/console-menu

**Comment by pranavbaburaj** [comment]
Hmm, but we will have to rewrite it entirely or reimplement it for windows 🤔

**Comment by GouravSardana** [comment]
Fow now should we only integrated it when OS is windows ?

**Comment by pranavbaburaj** [comment]
> Fow now should we only integrated it when OS is windows?

Yeah, We first check if the platform is windows, then create this interface for windows

**Comment by GouravSardana** [comment]
@pranavbaburaj Yes, This can be done but this would be temp. We have to migrate the whole things to something else

**Comment by Manas1820** [comment]
**There is a workaround to make it work on window**

```
if os.name == "nt":
import msvcrt
else:
import terminos
```

That is we can check if the current os the user is on is windows or not if we found its windows import [msvcrt](https://docs.python.org/3.7/library/msvcrt.html#console-i-o) and replace the same functionalities which are doing using terminos . @GouravSardana @pranavbaburaj this solution should work and looks like the only workaround rn.
